src/server.js

```diff
@@ -103,7 +104,42 @@ app.post('/api/parcelState/group', server.handleRequest(getParcelStateGroup))
 
 export function getParcelStateGroup(req) {
   const coordinates = server.extractFromReq(req, 'coordinates')
-  return ParcelState.findInCoordinates(coordinates)
+
+  // Point scan
+  if (coordinates.length <= env.get('PARCEL_RANGE_THRESHOLD', 100)) {
+    return ParcelState.findInCoordinates(coordinates)
+  }
+
+  // Range scan
+  const xCoords = []
+  const yCoords = []
+
+  coordinates.forEach(coord => {
+    const [x, y] = coordinatesUtils.toArray(coord)
+    xCoords.push(parseInt(x, 10))
+    yCoords.push(parseInt(y, 10))
+  })
+
+  const mincoords = [
+    Math.min.apply(null, xCoords),
+    Math.min.apply(null, yCoords)
+  ]
+  const maxcoords = [
+    Math.max.apply(null, xCoords),
+    Math.max.apply(null, yCoords)
+  ]
+
+  const isInQuery = parcel => {
+    for (const coord of coordinates) {
+      if (coord === parcel.id) {
+        return true
+      }
+    }
+    return false
+  }
+  return ParcelState.inRange(mincoords, maxcoords).then(parcels => {
+    return parcels.filter(parcel => isInQuery(parcel))
```

Comment:
Awesome!
last thing, I think you can replace isInQuery with: `coordinates.includes(parcel.id)` (I might be missing something)
So you can do `parcels.filter(parcel => coordinates.includes(parcel.id))`